osm2streets/src/operations/collapse_short_road.rs

```diff
@@ -96,6 +96,7 @@ impl StreetNetwork {
         // If the intersection types differ, upgrade the surviving interesting.
         if destroy_i.control == IntersectionControl::Signalled {
             self.intersections.get_mut(&keep_i).unwrap().control = IntersectionControl::Signalled;
+            // TODO Propagate to stop lines
```

Comment:
Ordering dataflow gets even more fun. `TrafficInterruption` and `IntersectionControl` are related.

And a fun case: https://www.openstreetmap.org/node/10249797837 is a `highway=stop` node right before a traffic signal. The mapper meant it as a stop line. But from reading the wiki, I'm not sure this is correct -- I think traffic signal nodes on the incoming ways would be more appropriate?